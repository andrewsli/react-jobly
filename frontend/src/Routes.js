import React, { Component } from 'react';
import { Route, Switch } from "react-router-dom";
import NotFound from "./NotFound";
import Home from "./Home";
import Companies from "./Companies";
import Jobs from "./Jobs";
import Profile from "./Profile";
import LoginSignup from "./LoginSignup";
import Company from "./Company";
import Logout from "./Logout";

class Routes extends Component {
  render() {
    const { updateUserDetails, loginUser, logOutUser } = this.props;
    return (
      <Switch>
        <Route exact path="/" render={() => <Home/>} />
        <Route exact path="/companies" render={() => <Companies />} />
        <Route exact path="/companies/:handle" render={(rtProps)=> <Company {...rtProps} updateCurrUser={updateUserDetails}/>} />
        <Route exact path="/jobs" render={() => <Jobs updateCurrUser={updateUserDetails}/>} />
        <Route exact path="/profile" render={() => <Profile updateCurrUser={updateUserDetails} />} />
        <Route exact path="/login" render={(rtProps) => <LoginSignup {...rtProps} loginUser={loginUser}/>} />
        <Route exact path="/logout" render={() => <Logout logOutUser={logOutUser}/>} />
        <Route render={() => <NotFound />}/>
      </Switch>
    );
  }
}

export default Routes;
