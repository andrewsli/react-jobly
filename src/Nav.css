.Nav {
    background-color: tomato;
    text-align: center;
    font-weight: bold;
  }
  
  .Nav a {
    padding: 0.5em 0;
    width: 10em;
    display: inline-block;
    color: white;
    text-decoration: none;
  }
  
  .Nav a:hover {
    background-color: black;
    color: white !important;
  }
  
  .Nav a.active {
    font-weight: bold;
    color: black;
  }
  