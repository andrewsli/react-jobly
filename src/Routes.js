import React, { Component } from 'react';
import { Route, Switch } from "react-router-dom";
import NotFound from "./NotFound";
import Home from "./Home";
import Companies from "./Companies";
import Jobs from "./Jobs";
import Profile from "./Profile";
import LoginSignup from "./LoginSignup";
import Company from "./Company";

class Routes extends Component {
  render() {
    return (
      <Switch>
        <Route exact path="/" render={() => <Home />} />
        <Route exact path="/companies" render={() => <Companies />} />
        <Route exact path="/companies/:handle" render={(rtProps)=> <Company {...rtProps} />} />
        <Route exact path="/jobs" render={() => <Jobs />} />
        <Route exact path="/profile" render={() => <Profile />} />
        <Route exact path="/login" render={() => <LoginSignup />} />
        <Route exact path="/logout" render={() => <Home />} />
        <Route render={() => <NotFound />}/>
        {/* Alternative for Not Found: <Redirect to="/" /> */}
      </Switch>
    );
  }
}

export default Routes;
