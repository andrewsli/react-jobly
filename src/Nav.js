import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import "./Nav.css";

class Nav extends Component {
  static defaultProps={
    loggedIn: false,
  }
  
  render() {
    return (
      <nav className="Nav">
        <NavLink exact to="/">Home</NavLink>
        <NavLink exact to="/companies">Companies</NavLink>
        <NavLink exact to="/jobs">Jobs</NavLink>
        <NavLink exact to="/profile">Profile</NavLink>
        {this.props.loggedIn
          ?
          <NavLink exact to="/logout">Logout</NavLink>
          :
          <NavLink exact to="/login">Login</NavLink>
        }
      </nav>
    );
  }
}

export default Nav;
