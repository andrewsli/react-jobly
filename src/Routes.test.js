import React from 'react';
import Nav from "./Nav";
import Routes from './Routes';
import toJson from "enzyme-to-json";
import { shallow, mount } from "enzyme";
import { MemoryRouter } from "react-router-dom"

it('renders without crashing using shallow', () => {
  shallow(<Routes />)
});

it('renders without crashing using mount', () => {
  mount(<MemoryRouter>
    <Routes />
  </MemoryRouter>);
});

it("matches snapshot", function () {
  let wrapper = shallow(<Routes />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

describe("should render component corresponding to url (unit)", function () {
  it("should go to home page with '/'", function () {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/", "/companies", "/jobs", "/profile", "/login", "/logout", "/notapage"]}
        initialIndex={0}>
        <Routes />
      </MemoryRouter>);

    expect(wrapper.find("Home").length).toBe(1);
  });

  it("should go to companies page with '/companies'", function () {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/", "/companies", "/jobs", "/profile", "/login", "/logout", "/notapage"]}
        initialIndex={1}>
        <Routes />
      </MemoryRouter>);

    expect(wrapper.find("Companies").length).toBe(1);
  });

  it("should go to jobs page with '/jobs'", function () {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/", "/companies", "/jobs", "/profile", "/login", "/logout", "/notapage"]}
        initialIndex={2}>
        <Routes />
      </MemoryRouter>);

    expect(wrapper.find("Jobs").length).toBe(1);
  });

  it("should go to user profile page with '/profile'", function () {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/", "/companies", "/jobs", "/profile", "/login", "/logout", "/notapage"]}
        initialIndex={3}>
        <Routes />
      </MemoryRouter>);

    expect(wrapper.find("Profile").length).toBe(1);
  });

  it("should go to profile page with '/login' when logged out", function () {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/", "/companies", "/jobs", "/profile", "/login", "/logout", "/notapage"]}
        initialIndex={4}>
        <Routes />
      </MemoryRouter>);

    expect(wrapper.find("LoginSignup").length).toBe(1);
  });

  it("should redirect to home page with /logout when logged in", function () {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/", "/companies", "/jobs", "/profile", "/login", "/logout", "/notapage"]}
        initialIndex={5}>\
        <Routes />
      </MemoryRouter>);

    expect(wrapper.find("Home").length).toBe(1);
  });


  it("should return not found with non existent page", function () {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/", "/companies", "/jobs", "/profile", "/login", "/logout", "/notapage"]}
        initialIndex={6}>
        <Routes />
      </MemoryRouter>);

    expect(wrapper.find("NotFound").length).toBe(1);
  });
})

describe("renders right component when link is clicked", function () {
  it("should go to home page when the logo in nav bar is clicked (integrated)",
    function () {
      const wrapper = mount(
        <MemoryRouter>
          <Nav />
          <Routes />
        </MemoryRouter>);

      wrapper
        .find('a[href="/"]')
        .simulate("click", { button: 0 });

      expect(wrapper.find("Home").length).toBe(1);
    });

  it("should go to the companies page when the companies tab is clicked",
    function () {
      const wrapper = mount(
      <MemoryRouter>
        <Nav />
        <Routes />
      </MemoryRouter>);

      wrapper
        .find('a[href="/companies"]')
        .simulate("click", { button: 0 });
      
      expect(wrapper.find("Companies")).toHaveLength(1);
    });

  it("should go to the jobs page when the jobs tab is clicked",
    function () {
      const wrapper = mount(
      <MemoryRouter>
        <Nav />
        <Routes />
      </MemoryRouter>);

      wrapper
        .find('a[href="/jobs"]')
        .simulate("click", { button: 0 });

      expect(wrapper.find("Jobs")).toHaveLength(1);
    });

  it("should go to the profile page when the profile tab is clicked",
    function () {
      const wrapper = mount(
      <MemoryRouter>
        <Nav />
        <Routes />
      </MemoryRouter>);

      wrapper
        .find('a[href="/profile"]')
        .simulate("click", { button: 0 });

      expect(wrapper.find("Profile")).toHaveLength(1);
    });

  it("should log out when the log out button is clicked",
    function () {
      const wrapper = mount(
      <MemoryRouter>
        <Nav loggedIn={true}/>
        <Routes />
      </MemoryRouter>);

      wrapper
        .find('a[href="/logout"]')
        .simulate("click", { button: 0 });

      expect(wrapper.find("Home")).toHaveLength(1);
    });

  it("should show the log in form when the log in button is clicked",
    function () {
      const wrapper = mount(
      <MemoryRouter>
        <Nav loggedIn={false}/>
        <Routes />
      </MemoryRouter>);

      wrapper
        .find('a[href="/login"]')
        .simulate("click", { button: 0 });
      
      expect(wrapper.find("LoginSignup")).toHaveLength(1);
    });
})
