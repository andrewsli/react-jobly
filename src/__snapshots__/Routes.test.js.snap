// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Switch>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/companies"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/jobs"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/profile"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/login"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/logout"
    render={[Function]}
  />
  <Route
    render={[Function]}
  />
</Switch>
`;
